packages/form-participation/test-web/MixinWithValidators.lit.test.ts

```diff
@@ -0,0 +1,169 @@
+import { elementUpdated, expect, fixture, fixtureCleanup } from '@open-wc/testing';
+import { FormControlMixin, Validator } from '../src';
+import { html, LitElement } from 'lit';
+import { customElement, query } from 'lit/decorators.js';
+import { spy } from 'sinon';
+
+const requiredValidator: Validator = {
+  attribute: 'required',
+  key: 'valueMissing',
+  message: 'This field is required',
+  callback(instance, value): boolean {
+    return !!value;
+  }
+};
+
+const otherValidator: Validator = {
+  attribute: 'other',
+  key: 'customError',
+  message: 'will not trigger',
+  callback() { return true; }
+}
+
+@customElement('form-control-lit-validators')
+class FormControlLitValidators extends FormControlMixin(LitElement) {
+  static get formControlValidators(): Validator[] {
+    return [requiredValidator, otherValidator];
+  }
+
+  @query('div')
+  validationTarget: HTMLElement;
+
+  render() {
+    return html`<div></div>`;
+  }
+
+  _onInput({ target }: Event & { target: HTMLInputElement }) {
+    this.value = target.value;
+  }
+}
+
+@customElement('with-checked')
+class FormControlLitCheckedValidators extends FormControlLitValidators {
+  checked = false;
+  value = 'foo';
+}
+
+describe('The FormControlMixin using LitElement', () => {
+  let form: Element;
+  let el: FormControlLitValidators;
+  let constructor: typeof FormControlLitValidators;
+
+  beforeEach(async () => {
+    form = await fixture(html`
+      <form>
+        <form-control-lit-validators
+          name="formControl"
+        ></form-control-lit-validators>
+      </form>
+    `);
+    // @ts-ignore
+    el = form.querySelector<FormControlLitValidators>('form-control-lit-validators');
+    constructor = el.constructor as typeof FormControlLitValidators;
+  });
+
+  afterEach(fixtureCleanup);
+
+  it('keeps a reference to validators', async () => {
+    expect(constructor.formControlValidators.length).to.equal(2);
+  });
+
+  it('will track the observedAttributes relative to validators', async () => {
+    expect(constructor.observedAttributes.includes('other')).to.be.true;
+  });
+
+  it('will return the validator from getValidator', async () => {
+    // @ts-ignore
+    expect(constructor.getValidator('required')).to.equal(requiredValidator);
+  });
+
+  it('will add validator attributes to observedAttributes', async () => {
+    expect(constructor.observedAttributes.includes('required')).to.be.true;
+  });
+
+  it('will mark the validity state as invalid given the callback', async () => {
+    expect(el.validity.valid).to.be.false;
+  });
+
+  it('will have a reference to the parent form', async () => {
+    expect(el.form).to.equal(form);
+  });
+
+  it('will not recommend showing an error when pristine', async () => {
+    expect(el.validity.valid).to.be.false;
```

Comment:
should there be an expect here making sure that its actually in the pristine state when the other expects are called?

Reply:
Not other than the `fixtureCleanup` this really is dependent on how the example was coded. Maybe not the best test in the world.